kubernetes/cache/proxy_status.go

```diff
@@ -18,46 +19,51 @@ type ProxyStatusCache interface {
 // pollIstiodForProxyStatus is a long running goroutine that will periodically poll istiod for proxy status.
 // Polling stops when the stopCacheChan is closed.
 func (c *kialiCacheImpl) pollIstiodForProxyStatus(ctx context.Context) {
-	log.Debug("[Kiali Cache] Starting polling istiod for proxy status")
-	go func() {
-		for {
-			select {
-			case <-ctx.Done():
-				log.Debug("[Kiali Cache] Stopping polling for istiod proxy status")
-				return
-			case <-time.After(c.tokenNamespaceDuration):
-				// Get the proxy status from istiod with some retries.
-				// Wrapping this so we can defer cancel.
-				func() {
-					ctx, cancel := context.WithTimeout(ctx, c.tokenNamespaceDuration)
-					defer cancel()
+	if config.Get().ExternalServices.Istio.IstioAPIEnabled {
```

Comment:
Can you move this if statement up to where the function is called in `NewKialiCache` https://github.com/kiali/kiali/blob/master/kubernetes/cache/cache.go#L210-L211? Calling a function that actually does nothing if you've enabled a config option is an odd side effect and could be confusing later on.